books/models.py

```diff
@@ -147,3 +149,16 @@ def __unicode__(self):
     @models.permalink
     def get_absolute_url(self):
         return 'pathagar.books.views.book_detail', [self.pk]
+
+
+@receiver(post_delete, sender=Book)
+def book_post_delete_handler(**kwargs):
+    book = kwargs['instance']
+
+    book_storage, book_file = book.book_file.storage, book.book_file.path
+    if book_file is not None:
+        book_storage.delete(book_file)
+
+    cover_storage, cover_file = book.cover_img.storage, book.cover_img.path
+    if cover_file is not None:
```

Comment:
Smallish nitpick: usually, when you want to compare with an *"""empty"""* value, is usually sufficient to just do
`if foo:`. It is not 100% the same as `is not None` (the suggested comparison will be `True` if the value is an empty string, empty list, dict, and other similar cases, while the explicit `is not None` won't be), but as a general rule most of the time you want to just remove the `is not None` part.